py/bitbox02/Makefile

```diff
@@ -21,21 +21,38 @@ TARGETS=$(addprefix ${OUT_DIR}/, $(PROTO_FILES:.proto=_pb2.py))
 
 all: ${TARGETS}
 
+# To avoid protobuf name collisions with other packages used elsewhere like the
+# electrum app, the pb files for python are generated from a copy
+# of ../../messages where all protos specify a non-empty package directive.
+# The proto files need not be committed to the repo. They are created at build time.
+PY_PROTO_DIR=.proto
+PY_PROTO=$(addprefix .proto/,${PROTO_FILES})
+
 # Explicit rules cannot have multiple outputs (it will be interpreted as multiple rules and
 # therefore the same command will be executed multiple times)
 # Instead we create a single implicit rule. Implicit rules allow multiple outputs without executing
 # the same command multiple times when multiple threads are used.
 # https://stackoverflow.com/questions/19822435/multiple-targets-from-one-recipe-and-parallel-execution
 TMP_TARGETS=$(addprefix %/, $(PROTO_FILES:.proto=_pb2.py))
-${TMP_TARGETS}: $(addprefix ../../messages/,${PROTO_FILES})
+${TMP_TARGETS}: ${PY_PROTO}
 	mkdir -p ${OUT_DIR}
-	protoc --proto_path=../../messages --python_out=${OUT_DIR} --mypy_out=${OUT_DIR} $^
+	protoc --proto_path=${PY_PROTO_DIR} --python_out=${OUT_DIR} --mypy_out=${OUT_DIR} $^
 	touch ${@D}/__init__.py
 # The hack below is due to protobuf-python generating absolute paths (which are frowned upon) instead of relative paths.
 # https://github.com/protocolbuffers/protobuf/issues/1491
 	sed -i 's/^from \([^.]*_pb2\)/from .\1/g'  ${OUT_DIR}/*.pyi
 	sed -i 's/^import \([^.]*_pb2\) as \([^.]*__pb2\)/from . import \1 as \2/g' ${OUT_DIR}/*.py
 
+# Ideally, the prerequisites would be only those which we need for PROTO_FILES
+# including their optional .options files, but it would make the rules more complicated.
+# So, we just grab all proto and options files, insert package name to protos
```

Comment:
you don't need the option files, they are only for nanopb. See 

https://jpa.kapsi.fi/nanopb/docs/concepts.html#modifying-generator-behaviour
https://jpa.kapsi.fi/nanopb/docs/reference.html#defining-the-options-in-a-options-file

I am not 100% sure but I think it only applies to nanopb and is not a general protobuf concept.

Reply:
They are also not copied to the Go lib, fyi.

Reply:
let me try without the options and see if it changes anything in pb2.py. never used options in separate files before.

Reply:
nice find! indeed, no changes in pb2.py files without options.
removed.

Reply:
I mean, options is a known feature of protobufs - https://developers.google.com/protocol-buffers/docs/proto3#options - but I've never used it in separate files. My favorite was `deprecated=true`, of course.

Reply:
So, if say you added `[deprecated=true]`, I'm pretty sure python would generate a pb2.py with a warning or something when that deprecated field is used. 

Not sure why options were split in a separate files and https://github.com/digitalbitbox/bitbox02-firmware/commit/ace1bc4d07fd4599c066bb9116bbb27ab4b72daf doesn't explain it either but we can always merge them back to .proto files if the protobuf usage becomes more complex like deprecating fields or adding custom options.

Reply:
It's also unclear what happens if you start adding options to both .proto and .options files.
Anyway, I think it's good as is for now.

Reply:
as i said, the separate file is a nanopb feature (i think ;)).

it is in a separate file because if you put it in the main file, the generators for Go/Python will fail (unrecognized options or sth like that).

Reply:
oh I thought you meant options in general is a nanopb feature.

> if you put it in the main file, the generators for Go/Python will fail (unrecognized options or sth like that).

I don't it would fail because you then specify which plugin it is, like in nanopb example:
 
```
message MyMessage {
    required string name = 1 [(nanopb).max_size = 40];
    repeated int32 ids = 4   [(nanopb).max_count = 5];
}
```

the `(nanopb)` tells protoc it's for nanopb generator.